qiskit/mapper/_coupling.py

```diff
@@ -87,14 +87,23 @@ def __init__(self, couplingdict=None):
         # self.dist is a dict of dicts from node pairs to distances
         # it must be computed, it is the distance on the digraph
         self.dist = None
+
+        if couplingdict is not None and couplinglist is not None:
+            raise CouplingError('Cannot specify both couplingdict and couplinglist')
+
         # Add edges to the graph if the couplingdict is present
         if couplingdict is not None:
```

Comment:
same as above i would not have this code.

Reply:
This can be removed when couplingdict is removed.